apps/client/src/app/components/user-account-settings/user-account-settings.component.ts

```diff
@@ -277,13 +281,30 @@ export class UserAccountSettingsComponent implements OnDestroy, OnInit {
       });
   }
 
-  private registerDevice() {
+  private registerDevice(aEvent: MatSlideToggleChange) {
```

Comment:
I would prefer to change this function to return a promise (and throwing an error) instead of wiring it with `MatSlideToggleChange`.

Reply:
Yup, that'd be another option.